crates/roc_host_lib/src/lib.rs

```diff
@@ -8,16 +7,19 @@ use std::borrow::Borrow;
 /// Note we use argc and argv to pass arguments to the program instead of std::env::args().
 #[no_mangle]
 pub unsafe extern "C" fn main(argc: usize, argv: *const *const i8) -> i32 {
-    let args = std::slice::from_raw_parts(argv, argc);
-
-    let mut args: RocList<ReadOnlyRocStr> = args
+    let args = std::slice::from_raw_parts(argv, argc)
         .iter()
         .map(|&c_ptr| {
             let c_str = std::ffi::CStr::from_ptr(c_ptr);
-            let roc_str = RocStr::from(c_str.to_string_lossy().borrow());
-            ReadOnlyRocStr::from(roc_str)
+
+            // TODO confirm this is ok... feels dangerous
+            let os_str =
+                std::ffi::OsString::from_encoded_bytes_unchecked(c_str.to_bytes().to_owned());
```

Comment:
Please read the docs and give me an opinion here 🙏

Reply:
> The byte encoding is an unspecified, platform-specific, self-synchronizing superset of UTF-8. By being a self-synchronizing superset of UTF-8, this encoding is also a superset of 7-bit ASCII.

So I think this is incorrect, but it also seems unnecessary. This seems like we could just pass the `&[u8]` to `ArgToAndFromHost::from(arg)` and handle it there. For Unix it's handled, and for Windows we'll need to do something special anyway.

Reply:
fixed in f6680903c